Objects/tupleobject.c

```diff
@@ -705,6 +705,26 @@ tuple_new_impl(PyTypeObject *type, PyObject *iterable)
         return PySequence_Tuple(iterable);
 }
 
+static PyObject *
+tuple_vectorcall(PyObject *type, PyObject * const*args,
+                 size_t nargsf, PyObject *kwnames)
+{
+    if (kwnames && PyTuple_GET_SIZE(kwnames) != 0) {
+        PyErr_Format(PyExc_TypeError, "tuple() takes no keyword arguments");
+        return NULL;
+    }
+    Py_ssize_t nargs = PyVectorcall_NARGS(nargsf);
+    if (nargs > 1) {
```

Comment:
I would be nice to have an helper function to raise the exception for use: function taking nargsf and calling PyVectorcall_NARGS() for us. Again, it should be done in a separated PR.

What does Argument Clinic use for that? Does it duplicate the code?

I'm also worried but the Python binary will contains tons of ""xxx takes no keyword arguments" duplicated strings, where only xxx changes. _PyArg_NoKeywords() builds a string from the function name and so doesn't have the issue.

Reply:
I agree with your all concerns :)
I will submit the PR to deal with!